independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenArtifactResolver.java

```diff
@@ -328,7 +373,7 @@ private CollectRequest newCollectManagedRequest(Artifact artifact, List<Dependen
         final ArtifactDescriptorResult descr = resolveDescriptor(artifact);
         Collection<String> excluded;
         if(excludedScopes.length == 0) {
-            excluded = Arrays.asList(new String[] {"test", "provided"});
+            excluded = Collections.emptyList();
```

Comment:
I agree this was an ugly choice to design it this way. But by default "test" and "provided" should be excluded. That is, if the caller doesn't provide anything.

Reply:
This is why there is now explicit devmode and test flags, as if you are running the dev mode tests you need both.

Reply:
That is why I changed this, as there are cases where there are no excluded scopes, so now it is up to the caller.

Reply:
This is a different default behavior though. But let's leave it for now. I didn't do it well in the first place, so I'll probably review it later.